### ISSUE TITLE
Modal dialog focus trap doesn't work when tabbing from browser navigation bar

### ISSUE THREAD

**Issue by vitalysoftech** [primefaces/primevue#8015]
### Describe the bug

Hello,

modal dialog is assumed to trap keyboard focus. It works right after you open a dialog. But try to put your keyboard focus to browser's navigation bar (the place where URL is entered) and start moving focus by pressing Tab key. You will notice that after you pass all focusable elements from the navigation bar, focus ring will move behind dialog mask. It breaks modality, so user may accidentally show another modal dialog or run code that is incompatible with currently running flow in shown dialog.

Regards,
Vitaly

### Reason for not contributing a PR

- [ ] Lack of time
- [x] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/primevue-4-vite-issue-template-anx9vgdr

### Environment

Any environment.

### Vue version

3.5.17

### PrimeVue version

4.3.7

### Steps to reproduce the behavior

1. Open the page: https://primevue.org/dialog/ or stackblitz sample.
2. Click on the "Show" button to show a modal dialog.
3. Place cursor to the browser's navigation bar.
4. Start tabbing until focus ring will appear behind dialog's mask.

![Image](https://github.com/user-attachments/assets/cb14b26f-9b1e-42b0-99bd-fa5be285d8b8)

### Expected behavior

Once focusable elements on the navigation bar are over, the focus must go directly to the top most modal dialog.

**Comment by tugcekucukoglu** [comment]
Which browser can I ask? Chrome, Safari and Firefox are behaving as expected.

**Comment by vitalysoftech** [comment]
Hi! I've just reproduced this in Chrome, Firefox and Edge. I've recorded another GIF video right from stackblitz. To reproduce you need to show dialog and set focus to URL box and then start tabbing. Notice in the end of the video how I can activate focused "Show" button multiple times pressing a Space key.

![Image](https://github.com/user-attachments/assets/056d8624-14b8-456e-a18b-38e0bbbf80fb)

**Comment by FrancisMawn** [comment]
We noticed the same issue using v4.3.7
